daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/NamedEntity.scala

```diff
@@ -33,7 +31,6 @@ object NamedEntity {
   final case class NRecDef(
       module: NModDef,
       name: DottedName,
-      dfn: Ast.DDataType,
```

Comment:
These `NamedEntity` types seem to replicate some parts of the AST, but these aren't used for anything, so I removed them. They're not needed in the name collision checker, which is the only thing these types are used for.

Reply:
:see_no_evil: